include/seqan3/argument_parser/validators.hpp

```diff
@@ -275,10 +297,24 @@ class file_ext_validator
     void operator()(std::filesystem::path const & path) const
     {
         std::string ext{path.extension().string()};
-        ext = ext.substr(std::min(1, static_cast<int>(ext.size()))); // drop '.' if extension is non-empty
-        if (!(std::find(extensions.begin(), extensions.end(), ext) != extensions.end()))
-            throw parser_invalid_argument(detail::to_string("Extension ", ext, " is not one of ",
-                                                            std::view::all(extensions), "."));
+        ext = ext.substr(std::min(1, std::max(0,static_cast<int>(ext.size())-1))); // drop '.' if extension is non-empty
+
+        // Case sensitive comparison
+        if (case_sensitive)
```

Comment:
You can reduce some duplicated code :) 
```cpp
std::string ext_tmp;

if (case_sensitive)
    ext_tmp = ext;
else
    ext_tmp = std::string{ext | view::to_lower}

if (!(std::find(extensions.begin(), extensions.end(), ext_tmp) != extensions.end()))
    throw parser_invalid_argument(detail::to_string("Extension ", ext, " is not one of ",                                                             
                                  std::view::all(extensions), "."));

```

Reply:
I can reduce even more and omit the tmp variable ;)
I think like this the error message is even mor intuitive because you can directly see what was internally compared:

```
        if ( !case_sensitive )
            ext = std::string{ext | view::to_lower};

        if (!(std::find(extensions.begin(), extensions.end(), ext) != extensions.end()))
                throw parser_invalid_argument(detail::to_string("Extension ", ext, " is not one of ",
                                                                std::view::all(extensions), "."));
```